user.go

```diff
@@ -39,6 +42,58 @@ func (a *App) GetUser(userID string) (*User, error) {
 	}
 }
 
+// GetUserByIdentifier gets a user using their identifier
+// returns user on success, error on failure
+func (a *App) GetUserByIdentifier(identifier string) (*User, error) {
+	var errorText string
+	message := "failed to get Passage User By Identifier"
+	limit := 1
+	res, err := a.client.ListPaginatedUsersWithResponse(
+		context.Background(),
+		a.ID,
+		&ListPaginatedUsersParams{
+			Limit:      &limit,
+			Identifier: &identifier,
+		},
+	)
+
+	if err != nil {
+		return nil, Error{Message: "network error:failed to get Passage User by Identifier"}
+	}
+
+	if res.JSON200 != nil {
+		users := res.JSON200.Users
+		if len(users) == 0 {
+			message = fmt.Sprintf(IdentifierDoesNotExist, identifier)
+			return nil, Error{
+				Message:    message,
+				StatusCode: 404,
+				StatusText: "404 not found",
+				ErrorText:  "user_not_found",
+			}
+		}
+
+		return a.GetUser(users[0].ID)
+	}
+
+	switch {
+	case res.JSON401 != nil:
+		errorText = res.JSON401.Error
+	case res.JSON404 != nil:
+		errorText = res.JSON404.Error
+		message = fmt.Sprintf(IdentifierDoesNotExist, identifier)
```

Comment:
A 404 when listing would indicate there's something wrong with the app ID.
If no user is found, a 200 with an empty result array will be returned instead.

So the message modification here isn't needed, as it's already used above [here](https://github.com/passageidentity/passage-go/pull/53/files#diff-40377b7a7613e54bce00e5f123d734c2a159862c936abae1c69305f5dad35eafR67).
```suggestion
```

Reply:
good call out! Made these changes